### PR TITLE
Featurizer Rewrite

### PR DESCRIPTION
**Pull request by agude**
These commits begin to add a HOG featurizer to Pelops.

## Feedback Needed

I'd like some feedback on `FeatureProducer()` which will probably be the API for this. I had to mangle it a lot compared to its previous version which no longer works with the `ChipDataset()` class we now have. (For example, it used to have a double for loop, but `ChipDataset()` now yields chips instead of a list of chips.)

Currently the idea is roughly as follows:

Each child class implements its own `produce_features(self, chip)` function that takes a single chip and returns an array of features. They also implement `set_variables(self)` that set whatever internal stuff they need, so far this is only self.feat_size to tell the class how large their feature vectors are.

## ResNet50, and Fine-tuning

We will also need to implement a ResNet50 featurizer using this class as well, and the fine-tuned networks will also have to slot in. Does this need extension to handle those cases?

## TODO

- [ ] Add tests
- [x] Add ResNet50 back 
- [ ] Make sure we write out `FeatureDatasets`

### CODE REVIEW

**Comment by d-grossman** [comment]
here seem to be some calls to functions that are not specific to HOG..
image_features
load_image
load_array
extract_feats_from_chips -> save feats from chips, 
    why the return True?

FeatureProducer class -> own file

in HOGFeatureProducer.produce_features
should the constants be passed in or are they really constant.  is the value of 16 different in lines 123,124? possibly move to some class of constants.

**Comment by agude** [comment]
Agree. I've cleaned out those functions now, they were cruft from the old code. Possible they'll need to be re-implemented, but we should do that as needed instead of keeping them all around.

**Comment by ymt123** [comment]
Sorry I'm late to the party here. I generally like the idea of a refactor but I think the resnet features still needs to be implemented (I might be biased since I was using those functions) before this should be merged. I think it's important since it will end up using many of the functions you are currently removing

File: pelops/etl/extract_feats_from_chips.py [hunk 1]
```diff
@@ -1,91 +0,0 @@
...
-        img: A preprocessed image of the correct dimensions
-        model: a
-
-    Returns:
-
-    """
-    features = model.predict(img)
-    return features
-
-def extract_feats_from_chips(chipdataset, output_fname):
```
Comment (ymt123) [inline pelops/etl/extract_feats_from_chips.py]: One thing we've lost in the pull request is creating a "FeatureDataset". Do we still believe that is the right abstraction for downstream tests?

Comment (agude) [inline pelops/etl/extract_feats_from_chips.py]: I think it is still, yes.

**Pull request merged**